orbit/pkg/update/flag_runner.go

```diff
@@ -0,0 +1,186 @@
+package update
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/fleetdm/fleet/v4/orbit/pkg/constant"
+	"github.com/fleetdm/fleet/v4/server/service"
+	"github.com/rs/zerolog/log"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"time"
+)
+
+// FlagRunner is a specialized runner to periodically check and update flags from Fleet
+// It is designed with Execute and Interrupt functions to be compatible with oklog/run
+//
+// It uses an OrbitClient, along with FlagUpdateOptions to connect to Fleet
+type FlagRunner struct {
+	orbitClient *service.OrbitClient
+	opt         FlagUpdateOptions
+	cancel      chan struct{}
+}
+
+// FlagUpdateOptions is options provided for the flag update runner
+type FlagUpdateOptions struct {
+	// CheckInterval is the interval to check for updates
+	CheckInterval time.Duration
+	// RootDir is the root directory for orbit state
+	RootDir string
+	// OrbitNodeKey is the orbit node key for the enrolled host
+	OrbitNodeKey string
+}
+
+// NewFlagRunner creates a new runner with provided options
+// The runner must be started with Execute
+func NewFlagRunner(orbitClient *service.OrbitClient, opt FlagUpdateOptions) (*FlagRunner, error) {
+	r := &FlagRunner{
+		orbitClient: orbitClient,
+		opt:         opt,
+		cancel:      make(chan struct{}, 1),
+	}
+	return r, nil
+}
+
+// Execute starts the loop checking for updates
+func (r *FlagRunner) Execute() error {
+	log.Debug().Msg("starting flag updater")
+
+	ticker := time.NewTicker(r.opt.CheckInterval)
+	defer ticker.Stop()
+
+	// Run until cancel or returning an error
+	for {
+		select {
+		case <-r.cancel:
+			return nil
+		case <-ticker.C:
+			log.Info().Msg("calling flags update")
+			didUpdate, err := r.DoFlagsUpdate()
+			if err != nil {
+				log.Info().Err(err).Msg("flags updates failed")
+			}
+			if didUpdate {
+				log.Info().Msg("flags updated, exiting")
+				return nil
+			}
+		}
+	}
+}
+
+// Interrupt is the oklog/run interrupt method that stops orbit when interrupt is received
+func (r *FlagRunner) Interrupt(err error) {
+	close(r.cancel)
+	log.Debug().Err(err).Msg("interrupt for flags updater")
+}
+
+// DoFlagsUpdate checks for update of flags from Fleet
+// It gets the flags from the Fleet server, and compares them to locally stored flagfile (if it exists)
+// If the flag comparison from disk and server are not equal, it writes the flags to disk, and returns true
+func (r *FlagRunner) DoFlagsUpdate() (bool, error) {
+	flagFileExists := true
+
+	// first off try and read osquery.flags from disk
+	osqueryFlagMapFromFile, err := readFlagFile(r.opt.RootDir)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return false, err
+		}
+		// flag file may not exist on disk on first "boot"
+		flagFileExists = false
+	}
+
+	// next GetConfig from Fleet API
+	flagsJSON, err := r.orbitClient.GetConfig(r.opt.OrbitNodeKey)
+	if err != nil {
+		return false, fmt.Errorf("error getting flags from fleet %w", err)
+	}
```

Comment:
Am getting the following error in Orbit (every 30s):
```
184:2022-09-20T10:42:43-03:00 INF flags updates failed error="error parsing flags unexpected end of JSON input"
```
Let's check for `len(flagsJSON) == 0` and do nothing (the default scenario of a user not setting the `command_line_flags` in Fleet).